app/reports/views.py

```diff
@@ -13,28 +24,146 @@ def view_reports():
+        abort(403)
+
+    form = EditIncidentReportForm()
+
+    if form.validate_on_submit():
+        report.vehicle_id = form.vehicle_id.data
+        report.license_plate = form.license_plate.data
+
+        # reformat location data - TODO error-checking for bad addresses
+        lat, lng = geocode(form.location.data)
+        report.location.latitude, report.location.longitude = lat, lng
+        report.location.original_user_text = form.location.data
+
+        d, t = form.date.data, form.time.data
+        report.date = datetime(year=d.year, month=d.month, day=d.day,
+                               hour=t.hour, minute=t.minute, second=t.second)
+
+        report.duration = parse_timedelta(form.duration.data)
+        report.agency = form.agency.data
+
+        # TODO upload picture_file
+        report.picture_url = form.picture_url.data
+        report.description = form.description.data
+
+        db.session.add(report)
+        db.session.commit()
+        flash('Report information updated.', 'form-success')
+    elif form.errors.items():
+        flash_errors(form)
+
+    # pre-populate form
+    form.vehicle_id.default = report.vehicle_id
+    form.license_plate.default = report.license_plate
+    form.bus_number.default = report.bus_number
+    form.led_screen_number.default = report.led_screen_number
+    form.location.default = report.location.original_user_text
+
+    form.date.default = report.date
+    form.time.default = report.date
+
+    form.duration.default = report.duration
+    form.agency.default = report.agency
+    form.picture_url.default = report.picture_url
+    form.description.default = report.description
+    form.process()
+
+    return render_template('reports/manage_report.html', report=report,
+                           form=form)
+
+
+@reports.route('/<int:report_id>/delete')
+@login_required
+def delete_report_request(report_id):
+    """Request deletion of a report."""
+    report = IncidentReport.query.filter_by(id=report_id).first()
+
+    if report is None:
+        abort(404)
+
+    """Either the user is deleting their own report, or the user is an admin.
+    Agency workers cannot delete reports for their own agency."""
+    if (report.user_id != current_user.id) and (not current_user.is_admin()):
+        abort(403)
+
+    return render_template('reports/manage_report.html', report=report)
+
+
+@reports.route('/<int:report_id>/_delete')
+@login_required
+def delete_report(report_id):
+    """Delete a report"""
+
+    report = IncidentReport.query.filter_by(id=report_id).first()
+    report_user_id = report.user_id
+
+    db.session.delete(report)
+    db.session.commit()
+    flash('Successfully deleted report.', 'success')
 
-    return render_template('reports/reports.html', reports=reports,
-                           agencies=agencies, show_dropdown=False)
+    # TODO - address edge case where an admin clicks on their own report from
```

Comment:
re: admin redirect to `/all` and not `/my-reports` - i actually don’t know, i left the TODO in there cuz i’m not sure. one of the reasons it took me an extra day to do this was because my first approach involved passing in a boolean variable, `user_only_reports`, that would lmk whether or not we were looking at a report that belonged to the current user. but i couldn’t find a way to do this without passing it thru the URL somehow (e.g. `/reports/13/True/info`), which a pretty glaring security risk. I realized a better way to do it would be to check the report’s user id instead, so I changed it, but am left with the edge case.

Do you know if there’s a way to pass a variable between routes without it being tied to the URL? 
For context, in `manage_report.html`, we link to a specific report from the `/all` or `/my-reports` route using this line:
`{% set href = url_for(route, report_id=report.id) %}` - I could pass in a variable here to signal which route we came from, but I’m not sure how to do that without it being tied to the url (which can easily be modified and bypassed).
There probably is a way but I do not know it. LMK if you can think of anything!